tests/executables_src/testIntcHandler.cpp

```diff
@@ -0,0 +1,376 @@
+  device.activateAsyncRead();
+
+  // Create the domain and check that it is active
+  auto accessor3 = device.getVoidRegisterAccessor("!3", {AccessMode::wait_for_new_data});
+  readWithTimeout(accessor3); // initial value
+  auto dummyInterruptTrigger =
+      device.getVoidRegisterAccessor("DUMMY_INTERRUPT_3"); // for triggering the dummy interrupt
+  dummyInterruptTrigger.write();
+  readWithTimeout(accessor3); // initial value
+
+  // Test 1: get an accessor for a sub SubDomain of MuxedInterruptDistibutor [3], which is in the already
+  // active SubDomain [3].
+  auto acc3_0 = device.getVoidRegisterAccessor("!3:0", {AccessMode::wait_for_new_data});
+  readWithTimeout(acc3_0);
+
+  // Test that the MuxedInterruptDistibutor [3] has been activated
+  BOOST_TEST(device.read<int32_t>("TEST3/MER") == 0x3); // MER always has two bits which both have to be set
+
+  // Test that the handshake for sub-domain [3, 0] has been activated
+  BOOST_TEST(device.read<int32_t>("TEST3/IER") == (1 << 0));
+
+  // Test 2: The the SubDomain behind the MuxedInterruptDistributor [3] itself is activated (not only the handshake) if
+  // the distributor is already active. We use SubDomain [3, 1] for this and test the SubDomain activation indirectly by
+  // the activation of the MuxedInterruptDistibutor  [3, 1]
+  auto acc3_1 = device.getVoidRegisterAccessor("!3:1:3", {AccessMode::wait_for_new_data});
+  readWithTimeout(acc3_1);
+
+  BOOST_TEST(device.read<int32_t>("TEST3/SUB1/MER") == 0x3);
+  BOOST_TEST(device.read<int32_t>("TEST3/SUB1/IER") == (1 << 3));
+}
+
+/*********************************************************************************************************************/
+/* probably this is the first one to be implemented
+ * if neither (SIE and CIE) nor MIR are present, or only IER is there, it writes 1<<N to IER and clears with 1<<N
+ * if neither ICR nor IAR are present: INTC writes the according bit mask to ISR
+ */
+BOOST_AUTO_TEST_CASE(testIERwithISR) { // TEST0
+  ChimeraTK::Device device;
+  device.open(std::string{testCdd});
+
+  BOOST_TEST(device.isOpened() == true);
+  auto accInterrupt =
+      device.getVoidRegisterAccessor("!0:4", {ChimeraTK::AccessMode::wait_for_new_data}); // See DAQ_READY
+  auto dummyInterrupt = device.getVoidRegisterAccessor("DUMMY_INTERRUPT_0"); // for triggering the dummy interrupt
+
+  device.activateAsyncRead();
+
+  BOOST_TEST(readWithTimeout(accInterrupt)); // initial value after activation
+
+  auto ier = device.getScalarRegisterAccessor<int>("TEST0/IER");
+
+  ier.read();
+
+  BOOST_TEST(0x10 == ier);
+
+  auto isr = device.getScalarRegisterAccessor<uint32_t>("TEST0/ISR");
+  // prepare the status before sending the interrupt
+  // set one more bit to be sensitive to the handshake (need to see changes)
+  isr = 0x110;
+  isr.write();
+
+  dummyInterrupt.write();
+
+  // wait until interrupt handler is done
+  BOOST_TEST(readWithTimeout(accInterrupt));
+
+  isr.read();
+
+  BOOST_TEST(isr == 0x10);
```

Comment:
This test is only checking that the single bit is set. It is not testing that
* only one of two sub-interrupts is triggered if there is an accessor for both and only one bit is active
* both sub-interrupts are in the same dummy interrupt if both bits are on and there are two accessors

Reply:
done